components/board/config.go

```diff
@@ -35,18 +35,16 @@ func (config *I2CConfig) Validate(path string) error {
 	return nil
 }
 
-// AnalogConfig describes the configuration of an analog reader on a board.
-type AnalogConfig struct {
+// AnalogReaderConfig describes the configuration of an analog reader on a board.
+type AnalogReaderConfig struct {
 	Name              string `json:"name"`
-	Pin               string `json:"pin"`         // analog input pin on the ADC itself
-	SPIBus            string `json:"spi_bus"`     // name of the SPI bus (which is configured elsewhere in the config file)
-	ChipSelect        string `json:"chip_select"` // the CS line for the ADC chip, typically a pin number on the board
```

Comment:
Why did these disappear? If you're trying to get this to work on non-MCP300x chips, that sounds difficult. My instinct is to leave structure of the reader config alone, and just rename it.

Reply:
as part of this scope the Analog Config for MCP3008 chips is being moved to just pi and generic linux, and the new common board config is supposed to be more general.

Reply:
...so how do those boards know which SPI bus and chip select to use?

Reply:
wdym by those boards? afaik the only other board currently using this config is the numato, which has analog readers built in so doesn't need an external ADC.

Reply:
Sorry, I had forgotten that most boards switched to using a `MCP3008AnalogConfig` instead of a `AnalogReaderConfig`. My mistake!